vlog4j-core/src/test/java/org/semanticweb/vlog4j/core/model/AtomImplTest.java

```diff
@@ -29,78 +30,103 @@
 import org.junit.Test;
 import org.semanticweb.vlog4j.core.model.api.Atom;
 import org.semanticweb.vlog4j.core.model.api.Constant;
+import org.semanticweb.vlog4j.core.model.api.Predicate;
 import org.semanticweb.vlog4j.core.model.api.Variable;
 import org.semanticweb.vlog4j.core.model.impl.AtomImpl;
 import org.semanticweb.vlog4j.core.model.impl.Expressions;
+import org.semanticweb.vlog4j.core.model.impl.PredicateImpl;
 
 public class AtomImplTest {
 
 	@Test
 	public void testGetters() {
-		Variable x = Expressions.makeVariable("X");
-		Variable y = Expressions.makeVariable("Y");
-		Constant c = Expressions.makeConstant("c");
-		Constant d = Expressions.makeConstant("d");
-		Atom atom = Expressions.makeAtom("p", x, c, d, y);
+		final Variable x = Expressions.makeVariable("X");
+		final Variable y = Expressions.makeVariable("Y");
+		final Constant c = Expressions.makeConstant("c");
+		final Constant d = Expressions.makeConstant("d");
+		final Atom atom = Expressions.makeAtom("p", x, c, d, y);
 
-		Set<Variable> variables = new HashSet<Variable>();
+		final Set<Variable> variables = new HashSet<>();
 		variables.add(x);
 		variables.add(y);
 
-		assertEquals("p", atom.getPredicate());
+		assertEquals("p", atom.getPredicate().getName());
+		assertEquals(atom.getTerms().size(), atom.getPredicate().getArity());
+
 		assertEquals(variables, atom.getVariables());
 		assertEquals(Arrays.asList(x, c, d, y), atom.getTerms());
 	}
 
 	@Test
 	public void testEquals() {
-		Variable x = Expressions.makeVariable("X");
-		Constant c = Expressions.makeConstant("c");
-		Atom atom1 = Expressions.makeAtom("p", Arrays.asList(x, c));
-		Atom atom2 = Expressions.makeAtom("p", x, c);
-		Atom atom3 = new AtomImpl("q", Arrays.asList(x, c));
-		Atom atom4 = new AtomImpl("p", Arrays.asList(c, x));
+		final Variable x = Expressions.makeVariable("X");
+		final Constant c = Expressions.makeConstant("c");
+
+		final Predicate predicateP = new PredicateImpl("p", Arrays.asList(x, c).size());
```

Comment:
"Arrays.asList(x, c).size()" should be replaced by 2! (also below)

Reply:
done in the next commit